app/api/wallet/provider/mnemonic_encrypted.ts

```diff
@@ -15,17 +15,23 @@ const encryption = new PrivateKeyEncryption(new Scrypt(), numOfBytes => {
   return Buffer.from(bytes)
 })
 
+interface PromptInterface {
+  prompt: PromptPassphrase
+}
+
 function initProvider (
   data: WalletPersistenceData<EncryptedProviderData>,
   network: EnvironmentNetwork,
-  promptPassphrase: PromptPassphrase
+  promptInterface: PromptInterface
```

Comment:
Hold the promise in another interface. The original implementation `PromptPassphrase` promise executed when this initProvider run, it should only execute when the prompt actually happening.